### DISCUSSION TITLE
[Form] How to delete existed form values via form instance?

### DISCUSSION THREAD

**Discussion by imakou** [ant-design/ant-design#34344] [General]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Said I have a set of form values like
```
{
 f01:{name:"ABC", age:11},
 f02:{name:"CCC", age:21},
 f03:{name:"bbb", age:31},
}
```

Is there a way to remove `f01` from form value instance?
I tied to use `setFieldsValue()`, but it only does update, doesn't support remove/delete.

Please advise.

Thanks.

### What does the proposed API look like?
  const [form] = Form.useForm();
 form.deleteFieldValue(NamePath[]);

**Comment by imakou** [comment]
@afc163 ,
Does Antd's form have an API to delete fieldsValue?

Thanks

**Comment by afc163** [comment]
You can use `setFieldsValue` like `setState` in React.

**Reply by imakou** [reply]
Thanks for replying.
But `setFieldsValue` seems can only update current fieldsValue in form, it cannot be assigned new values.
I made a playground.
https://codesandbox.io/s/form-methods-antd-4-19-0-forked-7nusb3?file=/index.js

Could you please give hints how to delete `John` for example?

Thanks. @afc163

**Comment by imakou** [comment]
In this case in the playground, we have form values as 
```
{ 
  0: {
    name: "John",
    age: 12,
    id: 1001,
  }
1: {
    name: "Wick"
    age: 32
    id: 2002
  }
}
```

But there is no way to move `0` property.
Should we consider this as a bug?

**Comment by afc163** [comment]
You can look Form.List for this situation: https://ant.design/components/form/#components-form-demo-dynamic-form-item

**Reply by imakou** [reply]
@afc163 😊

我的情況是表單載入的時候會有需要一些preset要先在form裡面，同時這些被載入的preset還能要被更改/刪除。

能夠遵循SSOT原則是最好的，但是現在就「刪除」這件事無法達成，除非指派form item是被state生成且管理，否則單靠form instance似乎無法做到「刪除」這件事😭

**Reply by afc163** [reply]
你现在这个删除是要删除整列，还是只是清空两个 input 里的数据。

**Reply by imakou** [reply]
@afc163 
I’d like to delete entire row, let’s say John, age 12. How can we do for deleting this row by leverage formInstance(form hook). 

Thanks.

**Reply by chillyistkult** [reply]
Looks like it's not possible, which is unfortunately a real dealbreaker.
